src/graphmap.rs

```diff
@@ -57,11 +57,79 @@ pub type DiGraphMap<N, E> = GraphMap<N, E, Directed>;
 ///
 /// Depends on crate feature `graphmap` (default).
 #[derive(Clone)]
-pub struct GraphMap<N, E, Ty> {
+#[cfg_attr(
+    feature = "serde-1",
+    derive(Serialize, Deserialize),
+    serde(bound(
+        serialize = "N: serde::Serialize, E: serde::Serialize",
+        deserialize = "N: serde::Deserialize<'de>, E: serde::Deserialize<'de>"
+    ))
+)]
+pub struct GraphMap<N, E, Ty> where N: Eq + Hash {
```

Comment:
The implementation details - which fields are used, of graph map are internal, so it would be good to not expose them. Otherwise make clear in documentation, that serialization of GraphMap in particular, is not stable across semver incompatible versions of the library. Thanks. See other comments, a more deliberate serialization format - not derived - could be appropriate.